### ISSUE TITLE
Improvement 4: Source and configure some "asahi-overlay" packages from Asahi's PKGBUILDs

### ISSUE THREAD

**Issue by onethirtyfive** [tpwrules/nixos-m1#38]
**Contingent on #37. See any prior issues for more context. It's a sequence.**

## Perform limited processing of PKGBUILDs contents to prepare for edge kernel and GPU support

_I've actually already done most of this work. (Not on github just yet.)_

Let's use [PKGBUILDs](https://github.com/AsahiLinux/PKGBUILDs) as the source of truth for which version of asahi tools we're building from. We pick one digest (say `stable` branch) and whether we're `stable` or `edge` kernel, build a corresponding kernel config, mesa configuration, etc... and they should all play well with each other. This is also where GPU support becomes configurable.

I already have stable GPU support running off this approach. :)

Trying out a new version of asahi should be synonymous with "updating the PKGBUILDs pin." Sometimes a bit more for hand tweaks, but generally just that.

I am reapplying my working changes to this new, more disciplined refactor. It shouldn't be more than a day or two.

I hope slogging through my sequence of improvements was worth it. This was the point.

**Comment by onethirtyfive** [comment]
#34 -- so this shows up in the original issue, which has a tidy list of the whole sequence now. :+1:
